### ISSUE TITLE
Escape character is not escaped when displaying no such command message

### ISSUE THREAD

**Issue by Mm2PL** [fish-shell/fish-shell#5102]
### System and version info
Fish version: 2.7.1
$TERM: xterm
System: Ubuntu Linux

### Description
Escape character(\033) is not escaped when displaying the no such command message.

### How to reproduce:
 * Type `\033\[41mTEST` into the prompt and press enter
 * Notice the text will be colored

[Asciicast](https://asciinema.org/a/Tg6KZzMUSzom4dQGWwwuvb7di)

### What should happen
The shell should display `\033\[41mTEST` or `033[41mTEST`(like bash does)

### What happens
The shell displays `(RED BACKGROUND COLOR CODE)TEST: no such command`

**Comment by faho** [comment]
One issue here is that, for this particular error, we often don't print it.

That's because we use the OS-provided "command-not-found" program, if available.

**Comment by mqudsi** [comment]
Two things:

* I fixed "this" in 34130539d65c0113422c3f20cd7785d6c5aabb3b
* This _isn't_ the output of fish's default command not found handler. _This_ is what `__fish_default_command_not_found_handler` looks like:
```fish
fish: Unknown command '\e\[41mTEST'
```

Under Ubuntu, I see "this" bug only it's not fish that's responsible. The shell displays 
```
TEST: command not found
```
all in red. But that text does not come from fish, and that's not fish's default command not found handler that's causing this. (It comes from the external command-not-found helper that fish spawned.)

So, yes, fish had this bug (and it's now fixed). But, no, it's not fish that's responsible for the behavior you see there.
